[ADC-513] Vue Integration

vasswann opened the pull request:
adCAPTCHA Vue integration

vasswann committed:
Basic setup and the structure done for Vue integration.

vasswann committed:
Vue integration example vue test app is ready for test.

vasswann committed:
Removed comments

vasswann committed:
Example Test done, Testing working on it

vasswann committed:
adCAPTCHA vue testing is done and the readme files

vasswann committed:
Update released bu vite - Using happy-dom environment instead of jsdom

vasswann committed:
Minor fix deprication

address this deprecation warning,  update the Vite configuration to use ESM instead of CommonJS (CJS)

vasswann committed:
Update for .env file and .env.local has created.

Lucasarenaterry commented on examples/node/.env:
```diff
@@ -1 +1 @@
-API_KEY=
+API_KEY= "835e7672-4754-423b-b583-e99ab8d28021"
```
Change this so its in a .env.local

and put "Your api key here"

vasswann replied:
I created .env.local and the api key been removed from .env

Lucasarenaterry commented on examples/vue/.gitignore:
```diff
@@ -0,0 +1,27 @@
+*.sw?
+# Ignore .env files containing sensitive data
+.env.local
+.env.*.local
```
add empty line at the bottom

Lucasarenaterry commented on examples/vue/src/assets/background.jpg:
could you change this to use the adcaptcha logo instead:

https://www.notion.so/adcaptcha/67d0195ee5944561a932923e9a61eefd?v=cede034db8794a6c9439000a0a1765a0&pvs=4

vasswann replied:
The image is used for background.  As we talked about it, it's remained the same.

Lucasarenaterry commented on examples/vue/src/components/ContactForm.vue:
```diff
@@ -0,0 +1,175 @@
+    font-size: 0.9rem; 
+  }
+}
+</style>
```
Add empty line

Lucasarenaterry commented on examples/vue/src/components/Footer.vue:
```diff
@@ -0,0 +1,27 @@
+    font-size: 1rem;
+    margin-bottom: 1rem;
+  }
+  </style>
```
Add empty line

Lucasarenaterry commented on examples/vue/src/components/HeroSection.vue:
```diff
@@ -0,0 +1,93 @@
+    padding: 1.5rem;
+  }
+}
+  </style>
```
Add empty line

Lucasarenaterry commented on examples/vue/src/components/Navbar.vue:
```diff
@@ -0,0 +1,47 @@
+.navbar-scrolled {
+  background-color: rgba(8, 7, 7, 0.9); 
+}
+  </style>
```
Add empty line

Lucasarenaterry commented on examples/vue/src/components/ResponseMessage.vue:
```diff
@@ -0,0 +1,20 @@
+    margin-bottom: 1rem;
+    text-align: center;
+  }
+  </style>
```
Add empty line

Lucasarenaterry commented on examples/vue/src/style.css:
```diff
@@ -0,0 +1,27 @@
+
+html {
+  scroll-behavior: smooth;
+}
```
Add empty line

Lucasarenaterry commented on packages/vue/src/components/adCAPTCHA.vue:
```diff
@@ -0,0 +1,45 @@
+    };
+  },
+});
+</script>
```
Add empty line here

Lucasarenaterry commented on packages/vue/src/index.tsx:
```diff
@@ -0,0 +1,6 @@
+import AdCAPTCHA from './components/adCAPTCHA.vue';  
+import { setKeywords, getSuccessToken } from './util'; 
+export { AdCAPTCHA, setKeywords, getSuccessToken };
+export default AdCAPTCHA;
```
Remove comment and add empty line to the bottom

vasswann committed:
Minor Requested changes done.  packages/vue added to dependences

Lucasarenaterry commented on examples/vue/.gitignore:
```diff
@@ -0,0 +1,27 @@
+*.sw?
+# Ignore .env files containing sensitive data
+.env.local
+.env.*.local
```
.env.local
.env.*.local

These are not needed as you already have *.local

vasswann replied:
.gitignore update is done

vasswann committed:
Vue example .gitignore file updated

MattGibney commented on examples/vue/src/assets/background.jpg:
Remove this. The examples needs to be professional and on-brand with adCAPTCHA.

vasswann replied:
Background image removed, adCAPTCHA logo has added to the example page
<img width="1495" alt="Screenshot 2024-11-19 at 10 28 37" src="https://github.com/user-attachments/assets/b5925faf-fe42-4ca7-af91-546a0e180d0a">

MattGibney commented on examples/vue/src/components/Footer.vue:
We don't really need this, especially not a copyright.

vasswann replied:
Footer component removed

vasswann committed:
README.md file updated.

MattGibney commented on examples/vue/src/assets/background.jpg:
Remove this. Example styling needs to be brand consistent.

vasswann replied:
Image has been removed.

vasswann committed:
minor updates

Removed footer, removed hero section background image. adCAPTCHA logo added to the page.
